.github/workflows/update_docs.yml

```
@@ -0,0 +1,30 @@
+    branches:
+      - main
+      - add_docs
+
+jobs:
+  pages:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./docs
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)